exla/lib/exla.ex

```diff
@@ -355,6 +355,43 @@ defmodule EXLA do
     Nx.Defn.stream(function, args, Keyword.put(options, :compiler, EXLA))
   end
 
+  @doc """
+  Takes in a function, the templates variables and the compilation options
+  and returns the `EXLA.Executable` struct.
+
+  ## Examples
+
+      iex> fun = fn x, y -> Nx.add(Nx.sin(x), Nx.cos(y)) end
+      iex> args = [1.0, 2.0]
+      iex> module = EXLA.to_mlir_module(fun, args)
+      iex> EXLA.MLIR.Module.as_string(module)
+      ~c\"\"\"
+      module {
+        func.func public @main(%arg0: tensor<f32>, %arg1: tensor<f32>) -> tensor<f32> {
+          %0 = stablehlo.sine %arg0 : tensor<f32>
+          %1 = stablehlo.cosine %arg1 : tensor<f32>
+          %2 = stablehlo.add %0, %1 : tensor<f32>
+          stablehlo.return %2 : tensor<f32>
+        }
+      }
+      \"\"\"
+  """
+  def to_mlir_module(function, args, options \\ []) do
+    comp_fun = fn _key, callback ->
+      {:ok, {_xla_time, executable, _extra, _outfeed}} = callback.()
+      throw({:mlir_module, executable.ref})
```

Comment:
Gah, this is relying on internals of another module but I can't think of anything better for now, so ship it.